web-registry/src/lib/ecocredit/api.ts

```diff
@@ -290,22 +310,54 @@ const getClassProjectForBatch = async (
       } else {
         // TODO Keeping the old path for retro-compatibility of the function.
         // Once all paths use the react-query one we will be able to remove it.
-        metadata = await getMetadata(project.metadata, dataClient);
+        projectMetadata = await getMetadata(project.metadata, dataClient);
       }
     } catch (error) {}
   }
 
-  // TODO should we use credit class metadata instead of Sanity eventually?
-  // https://github.com/regen-network/regen-registry/issues/1428
   const creditClassSanity = findSanityCreditClass({
     sanityCreditClassData,
     creditClassIdOrUrl: project?.classId ?? '',
   });
 
+  const classId = project?.classId;
+  if (reactQueryClient) {
+    classData = await getFromCacheOrFetch({
+      query: getClassQuery({
+        request: { classId },
+        client: ecocreditClient,
+      }),
+      reactQueryClient,
+    });
+  } else {
+    // TODO Keeping the old path for retro-compatibility of the function.
```

Comment:
Here the task I was referring to for that: https://app.zenhub.com/workspaces/regen-registry-5f8998bec8958d000f4609e2/issues/gh/regen-network/regen-registry/1539